### PR TITLE
Improve ci initialization script

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes:https://github.com/NuGet/Client.Engineering/issues/1864

Regression? Last working version:

## Description
1. Cherry-pick https://github.com/NuGet/NuGet.Client/commit/9c349a023da147ba81a865f94ca1e52eb5ac028c commit into release-6.0.x branch. This is to fix a bug when build is triggered by "Approved for CI" label.

If the build is triggered by "Approved for CI" label, the "Configure VSTS CI Environment" step will fail to find the localization branch: [build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6695587&view=logs&j=d80b2849-30d4-52dd-74cd-d6536ba98fe0&t=64c204f9-1df6-57fc-6b34-ec5bd3372627&l=15)
```
Looking for branch 'head' in NuGet.Build.Localization
NuGet.Build.Localization Branch: head
git update NuGet.Build.Localization at D:\a\_work\1\s\submodules\NuGet.Build.Localization
fatal: couldn't find remote ref head
```

After cherry-pick this commit, the "Configure VSTS CI Environment" step will be like: [build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=6699945&view=logs&j=d80b2849-30d4-52dd-74cd-d6536ba98fe0&t=64c204f9-1df6-57fc-6b34-ec5bd3372627&l=14)
```
Looking for branch 'head' in NuGet.Build.Localization
NuGet.Build.Localization Branch: dev
git update NuGet.Build.Localization at D:\a\_work\1\s\submodules\NuGet.Build.Localization
```
2. Cherry-pick https://github.com/NuGet/NuGet.Client/commit/5d0416402856e247ae7c71d139f6ab8c2a4f26bd commit into release-6.0.x branch. This is to fix a bug that first cherry-pick needs target [GetNuGetSdkVsSemanticVersion](https://github.com/NuGet/NuGet.Client/blob/dev/scripts/cibuild/ConfigureVstsBuild.ps1#L175) to be defined. Otherwise, cherry-picking the first commit would fail running script ConfigureVstsBuild.ps1 at GetNuGetSdkVsSemanticVersion.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by zivkan — DISMISSED** [review]
You didn't fill out the checklist.

File: src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj [hunk 1]
```diff
@@ -9,6 +9,9 @@
     <Shipping>true</Shipping>
     <PackProject>true</PackProject>
     <IncludeInVSIX>true</IncludeInVSIX>
+    <!-- VS Extensibility APIs should use package version corresponding to VS version it targets, to reduce confusion. Also keep assembly version stable to minimize binding redirect issues. -->
+    <PackageVersion>$(NuGetSdkVsSemanticVersion)$(PreReleaseInformationVersion)</PackageVersion>
+    <AssemblyVersion>$(NuGetSdkVsSemanticVersion).0</AssemblyVersion>
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio/NuGet.VisualStudio.csproj]: This is going to change the package and assembly version scheme, which we should not do in a patch version of 6.0

File: src/NuGet.Clients/NuGet.VisualStudio.Contracts/NuGet.VisualStudio.Contracts.csproj [hunk 2]
```diff
@@ -13,6 +13,9 @@
     <IncludeNuGetSharedFiles>true</IncludeNuGetSharedFiles>
     <RootNamespace>NuGet.VisualStudio.Contracts</RootNamespace>
     <Description>RPC contracts for NuGet's Visual Studio Service Broker extensibility APIs.</Description>
+    <!-- VS Extensibility APIs should use package version corresponding to VS version it targets, to reduce confusion. Also keep assembly version stable to minimize binding redirect issues. -->
+    <PackageVersion>$(NuGetSdkVsSemanticVersion)$(PreReleaseInformationVersion)</PackageVersion>
+    <AssemblyVersion>$(NuGetSdkVsSemanticVersion).0</AssemblyVersion>
```
Comment (zivkan) [inline src/NuGet.Clients/NuGet.VisualStudio.Contracts/NuGet.VisualStudio.Contracts.csproj]: This is going to change the package and assembly version scheme, which we should not do in a patch version of 6.0

**Review by zivkan — DISMISSED** [review]

**Comment by zivkan** [comment]
@NuGet/nuget-client need reviews/approvals please

**Review by erdembayar — APPROVED** [review]
LGTM

**Review by jeffkl — APPROVED** [review]

**Pull request merged**